src/sage/groups/perm_gps/permgroup_named.py

```diff
@@ -3435,3 +3437,104 @@ def codegrees(self):
         ret = [self._m * i for i in reversed(range(self._n-1))]
         ret.append((self._n-1)*self._m - self._n)
         return tuple(sorted(ret, reverse=True))
+
+class SmallPermutationGroup(PermutationGroup_generic):
+    r"""
+    A GAP SmallGroup, returned as a permutation group.
+
+    GAP contains a library SGL of small groups, each identified by
+    its GAP SmallGroup id. (MAGMA uses the same identifiers).
+    The GAP SmallGroup id is a pair ``[n,k]`` consisting of
+    ``n``, the order of the group, and ``k``, an index determining
+    the group specifically. This class can construct the group as a
+    permutation group from this data.
+
+    INPUT:
+
+    - ``order`` -- the order of the group
+
+    - ``gap_id`` -- the numerical index in the GAP id of the group
+
+    Generators may be obtained through the :meth:`gens` method.
+    These could change for a particular group in later releases
+    of GAP. In many instances the degree of the constructed group
+    ``SmallPermutationGroup(n,k)`` will be a permutation group on
+    `n` letters, but this will not always be true.
+
+    EXAMPLES::
+
+        sage: G = SmallPermutationGroup(12,4); G
+        Group of order 12 and GAP Id 4 as a permutation group
+        sage: G.gens()
+        ((1,2)(3,5)(4,10)(6,8)(7,12)(9,11),
+        (1,3)(2,5)(4,7)(6,9)(8,11)(10,12),
+        (1,4,8)(2,6,10)(3,7,11)(5,9,12))
+        sage: G.character_table()
+        [ 1  1  1  1  1  1]
+        [ 1 -1 -1  1  1 -1]
+        [ 1 -1  1  1 -1  1]
+        [ 1  1 -1  1 -1 -1]
+        [ 2  0 -2 -1  0  1]
+        [ 2  0  2 -1  0 -1]
+        sage: all(SmallPermutationGroup(n,k).id()==[n,k] for n in [1..64] for k in [1..oeis(1)[n]])
```

Comment:
This test needs to be marked with `# optional - internet` and/or replaced by something that is manually inputted. (I would at least include part of the latter since it means it will get more regularly tested.)

Reply:
Why does it query data from OEIS that is part of the very small groups library that is  eing wrapped here in the first place,

Reply:
I replaced the oeis(1) by a call to libgap.NumberSmallGroups.